### ISSUE TITLE
Explore responsive iframe

### ISSUE THREAD

**Issue by RFSH** [informatics-isi-edu/chaise#1878]
Currently, we have to define a constant width and height for `iframe`s. This is not ideal and in some cases, we might want the iframe to be responsive based on the user's browser width. 

We should be able to do this, but it might require some extra changes in the way we're generating the `iframe` html. We should also make sure we're able to define min/max-width  and fully responsiveness might not the ideal case all the time.

**Comment by rastogi-bhavya** [comment]
The most popular way that I could find to do this was, to use a wrapper container around the `iframe`. 

http://jsfiddle.net/omarjuvera/8zkunqxy/2/

In this link as you can see the H & W of the iframe have been set to 100% of the parent. So when the size of the parent element changes the iframe also resizes. We can even add `min-width` and `max-width` to the parent element to make sure that the iframe does not go above and below a certain size.

**Comment by carlkesselman** [comment]
This looks good to me.

Carl

----------------------------------------------------------

Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute Viterbi School of Engineering Professor,

Preventive Medicine Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
 Email:  carl@isi.edu
Web: http://www.isi.edu/~carl
On Nov 30, 2020, 3:32 PM -0800, Bhavya Rastogi <notifications@github.com>, wrote:

The most popular way that I could find to do this was, to use a wrapper container around the iframe.

http://jsfiddle.net/omarjuvera/8zkunqxy/2/

In this link as you can see the H & W of the iframe have been set to 100% of the parent. So when the size of the parent element changes the iframe also resizes. We can even add min-width and max-width to the parent element to make sure that the iframe does not go above and below a certain size.

**Comment by jrchudy** [comment]
This issue has 4 examples of iframes used in different contexts.
https://github.com/informatics-isi-edu/chaise/issues/1961

**Comment by jrchudy** [comment]
Sample HTML:
```HTML
<figure class="embed-block -chaise-post-load" style="width: 100%; height: 100%;">
  <figcaption class="embed-caption" style="">no styles</figcaption>
  <iframe src="https://dev.rebuildingakidney.org/deriva-webapps/plot/?config=gudmap-todate-bar-no-dimensions&amp;hideNavbar=true" class="fill-available "></iframe>
</figure>

**Comment by jrchudy** [comment]
![image](https://user-images.githubusercontent.com/2932600/103712291-40f92a00-4f6e-11eb-8aa4-7872138aefcf.png)

When height and width are both undefined, iframe will stretch to available space.

**Comment by jrchudy** [comment]
![image](https://user-images.githubusercontent.com/2932600/103716740-9e927400-4f78-11eb-8d94-970f8d6b52ab.png)


![image](https://user-images.githubusercontent.com/2932600/103716659-760a7a00-4f78-11eb-82e9-9d2e02ae73a6.png)

**Comment by jrchudy** [comment]
So for the iframe, there are 3 areas (or dimensions): B--Browser, C--Cell, I--Iframe. Each with min_w, w, max_w and min_h, h, max_h
From the demo, it seems like by default: when there is no I dimension defined, it seems like Cw = Bw and Cmin_h = 2 lines. 
  - If Ih is defined, then Cmin_h = 1 line + max(Imin_h, Ih)  and Cmax_h = 1 line + min(Ih, Imax_h).  
  - If Iw is defined, the Cmin_w = max(Imin_w, Iw) and Cmax_w = min(Iw, Imax_w)
  
On the responsiveness, it seems like it's done based C and B:
  - If Cmin_w > B_w --> show horizontal scroll bar
  - If Cmin_w < B_w --> If Cmin_w is defined, try to fill Bw. If Cw and Cmax_w is defined, show content no wider than min(Cw, Cmax_w).

Best practice:
  - if you want a specific iframe display dimension, specific Iw and Ih. --> we should honor this.
  - Expand iframe width to browser width: define Imax_h and Imin_w +  Use chaise-fill
    - Imax_h will limit the responsiveness from continually stretching the iframe (and cell)
  - Long caption: define Imin_h --> though I am not sure how browser would behave if Bh is < Cmax_h
     - Cmax_h might be unnecessary since the cell will auto resize to accommodate the long caption (x lines) and the defined Imin_h (ok)
     - if Bh is < Cmax_h, a scrollbar will be present in the browser page to scroll down to the rest of the content

Params:
  - figureStyle or cellStyle for manipulating the cell? --> meaning deprecating iframeStyle which is confusing.

**Comment by jrchudy** [comment]
While exploring how we could apply the `chaise-autofill` class and other styles to some of our existing iframe [examples](https://dev.isrd.isi.edu/~jchudy/chaise/record/#71851/reference_schema:reference_values/RID=1SA), I encountered the following:
 - When using `height` and `chaise-autofill` together, the height is ignored because `chaise-autofill` applies a `height` style that overrides the `height`
 - `min-height` creates a default height, but responsive apps force this height to continue to grow
 - `min-height` and `max-height` together can be used to mitigate the responsiveness
 - which implies we should allow height to be set to override `chaise-autofill`
   - do we make `chaise-autofill` only apply to width?
   - 2 separate classes, `chaise-autofill-width` and `chaise-autofill-height`?
   - support 3 classes including `chaise-autofill` for both properties?
   
#### Other alternatives include adding styles differently:
Do not use `chaise-autofill` and set `height` or `min-height` with `width: 100%`.
```html
::: iframe [CAPTION](https://example.com){style="width: 100%;  min-width: 500px; min-height: 400px;"} \n:::

# OUTPUT:
<figure class="embed-block">
    <figcaption class="embed-caption">CAPTION</a></figcaption>
    <iframe src="https://example.com" style="width: 100%; min-width: 500px; min-height: 400px;"></iframe>
</figure>
```

Use `chaise-autofill` and suggest adding `height` into the `figure-style` property so that the autofill will have the iframe resize to the height of it's parent, `<figure>`.
```html
::: iframe [CAPTION](https://example.com){class=chaise-autofill style="min-width: 500px;" figure-style="height: 800px;"} \n:::

# OUTPUT:
<figure class="embed-block" style="height: 800px;">
    <figcaption class="embed-caption">CAPTION</a></figcaption>
    <iframe src="https://example.com" class="chaise-autofill" style="min-width: 500px;"></iframe>
</figure>
```

**Comment by jrchudy** [comment]
[This PR](https://github.com/informatics-isi-edu/ermrestjs/pull/865) was merged in ermrestJS adding more information to the documentation. The stated class added is included in chaise PR #2046 .
